GetWorkflows Docs (#141)
dbos-inc/dbos-docs: docs/api-reference/contexts.md

---

#### `handlerCtxt.getWorkflows`

```typescript
getWorkflows(input: GetWorkflowsInput): Promise<GetWorkflowsOutput>
```

> This function allows querying workflow history. It takes in as input an object describing which workflows to retrieve:

Reviewer: ```suggestion
This function allows querying workflow execution history. Its input is an object describing which workflows to retrieve:
```

```typescript
export interface GetWorkflowsInput {
  workflowName?: string; // The name of the workflow function
  authenticatedUser?: string; // The user who ran the workflow.
  startTime?: string; // Timestamp in RFC 3339 format
  endTime?: string; // Timestamp in RFC 3339 format

>   status?: "PENDING" | "SUCCESS" | "ERROR"; // The status of the workflow.

Reviewer: Soon will be a new status after we merge the DLQ PR

  applicationVersion?: string; // The application version that ran this workflow.

>   limit?: number; // Return up to this many workflows IDs. IDs are ordered by workflow creation time.

Reviewer: Mention that by default returns all workflows.

}
```

> It returns as output an object containing a list of the [UUIDs](../tutorials/idempotency-tutorial.md) of all retrieved workflows:

Reviewer: ```suggestion
It returns as output an object containing a list of the [UUIDs](../tutorials/idempotency-tutorial.md) of all retrieved workflows, ordered by workflow creation time:
```
